src/downshift.js

```diff
@@ -622,6 +622,11 @@ class Downshift extends Component {
         inputValue: this.props.itemToString(this.props.selectedItem),
       })
     }
+
+    if (this.props.highlightedIndex !== prevProps.highlightedIndex) {
+      this.setHighlightedIndex()
```

Comment:
Rather than calling `setHighlightedIndex`, I'd prefer to extract [this callback](https://github.com/arahansen/downshift/blob/2286e064cad768b43754c42b195430b0d5e8b09a/src/downshift.js#L133-L137) into a new function called "scroll highlighted item into view" and call that here. Makes it more explicit.

Reply:
I agree, that seems better! Will update.